### PR TITLE
[BugFix] Fix segment size in metadata cache

### PR DESCRIPTION
**Pull request by wyb**
When loading a segment, only the footer and column meta are loaded,
and indexes are loaded lazily when they are used.
So segment memory usage is inaccurate when the segment is loading
and inserted into the metadata cache.
Therefore, the segment memory usage needs to be updated when
the indexes are loading.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 31 / 36 (86.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/rowset/column_reader.h|0|1|00.00%|[158]|
|:large_blue_circle:|src/storage/rowset/zone_map_index.h|0|1|00.00%|[101]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|3|4|75.00%|[179]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|13|15|86.67%|[278, 279]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/zone_map_index.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 30 / 37 (81.08%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/rowset/column_reader.h|0|1|00.00%|[158]|
|:large_blue_circle:|src/storage/rowset/zone_map_index.h|0|1|00.00%|[101]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|12|17|70.59%|[278, 279, 539, 678, 965]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/zone_map_index.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet.cpp|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 63 / 67 (94.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|17|21|80.95%|[392, 406, 575, 576]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/tablet_schema.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|15|15|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/zone_map_index.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|17|17|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.cpp|1|1|100.00%|[]|

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -267,7 +266,18 @@ SegmentPtr TabletManager::lookup_segment(std::string_view key) {
 void TabletManager::cache_segment(std::string_view key, SegmentPtr segment) {
     auto mem_cost = segment->mem_usage();
     auto value = std::make_unique<CacheValue>(std::move(segment));
-    fill_metacache(key, value.release(), (int)mem_cost);
+    fill_metacache(key, value.release(), mem_cost);
+}
+
+void TabletManager::update_segment_cache_size(std::string_view key) {
```
Comment (kevincai) [inline be/src/storage/lake/tablet_manager.cpp:272]: looks like this is not only update the cache size but also redo the fill-metacache?

Comment (wyb) [reply be/src/storage/lake/tablet_manager.cpp:272]: yes, current lru cache does not support updating value size, so use refill to update.

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 63 / 67 (94.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/rowset/column_reader.cpp|17|21|80.95%|[392, 406, 575, 576]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/tablet_schema.h|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.cpp|15|15|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/zone_map_index.cpp|2|2|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|17|17|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/segment.h|4|4|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/ordinal_page_index.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bitmap_index_reader.cpp|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/rowset/bloom_filter_index_reader.cpp|1|1|100.00%|[]|

**Review by tracymacding — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**